hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3457,7 +3456,7 @@ func (r *HostedClusterReconciler) reconcileAWSOIDCDocuments(ctx context.Context,
 	client := &http.Client{Transport: transport}
 
 	for _, path := range oidcPublicDocumentPaths() {
-		requestURL := "https://" + hcp.Status.ControlPlaneEndpoint.Host + ":" + strconv.Itoa(int(hcp.Status.ControlPlaneEndpoint.Port)) + path
```

Comment:
I think the port might be different, the easiest way to get the correct url is to use `cfg.Host`